docker/base/lifecycle/pre-stop.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+GIT_STATUS=`git status -s`
+if [ -z "$GIT_STATUS" ] ; then
+  exit 0
+fi
+
+CURRENT_BRANCH=`git rev-parse --abbrev-ref HEAD`
+AUTOSAVE_BRANCH="auto-save-$JUPYTERHUB_USER-$CURRENT_BRANCH"
```

Comment:
can we add the commit-sha here? Then you would also know more specifically what the base was

Reply:
SHA added